addons/product/tests/test_product_pricelist.py

```diff
@@ -127,3 +136,53 @@ def test_10_calculation_price_of_products_pricelist(self):
 
         self.env.company.external_report_layout_id = self.env.ref('web.external_layout_standard').id
         test_reports.try_report_action(self.cr, self.uid, 'action_product_price_list', wiz_data=data_dict, context=ctx, our_module='product')
+
+    def test_20_price_different_currency_pricelist(self):
+        pricelist = self.ProductPricelist.create({
+            'name': 'Currency Pricelist',
+            'currency_id': self.new_currency.id,
+            'item_ids': [(0, 0, {
+                'compute_price': 'formula',
+                'base': 'list_price',
+                'price_surcharge': 100
+            })]
+        })
```

Comment:
You could create the pricelist in the setup, and only create the item in the specific tests ;)

Reply:
I did not wanted to do this as the `pricelists` are local to each test. They don't need to be used by other test methods.

Reply:
Ok then, don't modify it :)